src/campaigns/swapPromoCampaign.ts

```diff
@@ -23,16 +28,21 @@ export const swapsCampaignAction = async () => {
   mmkv.set(CampaignKey.swapsLaunch, true);
   setTimeout(() => {
     logger.log('triggering swap promo action');
+    analytics.track('Presented Feature Promo', { campaign: 'swaps_launch' });
 
     Navigation.handleAction(Routes.SWAPS_PROMO_SHEET, {});
   }, 1000);
 };
 
-export const swapsCampaignCheck = async (): Promise<boolean> => {
+export type SwapPromoCampaignExclusions = 'no_assets' | 'already_swapped';
```

Comment:
@benisgold  would prefer this be an enum so we can do type.exclusion for return values instead of just strings

Reply:
could event extend the generic type which may be nice

Reply:
personally i feel like an enum is overkill here since each of these strings are only used once and only within this file (and this should be the case for any exclusions for any campaign)

Reply:
im beefing with enums rn but i can be convinced if you have a strong opinion

Reply:
what's wrong with returning strings if it's type guarded?